### ISSUE TITLE
[Frontend] Incremental compiler fails to compile extension method calls

### ISSUE THREAD

**Issue by annagrin** [dart-lang/sdk#40351]
Example: 

- initial compilation:

```
      extension NumberParsing on String {
        int parseInt() {
          return int.parse(this);
        }
      }

      int global = 42;

      class C {
        C(int this.field, int this._field);

        static int staticField = 0;
        static int _staticField = 1;

        int _field;
        int field;

        int methodFieldAccess(int x) {
          /* compile expression happens here */
          return x + _field + _staticField;
        }

        Future<int> asyncMethod(int x) async {
          return x;
        }
      }
```

- compile expression

`"1234".parseInt()`

- result
Error: The method 'parseInt' isn't defined for the class 'String'

- expected
success

**Comment by jensjoha** [comment]
Should be fixed with 071210f7356e57448251404f6fb7d572d0649c1e but please verify on your end.

**Comment by annagrin** [comment]
Verified the fix, compile expression gives expected result. Thanks!
